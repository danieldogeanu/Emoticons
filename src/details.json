{
	"labels": [
		{
			"class": "face",
			"name": "Face"
		},
		{
			"class": "name",
			"name": "Emoticon Name"
		},
		{
			"class": "copy",
			"name": "Get"
		}
	],
	"author": {
		"name": "Daniel Dogeanu",
		"url": "https://danieldogeanu.com"
	},
	"social": [
		{
			"name": "Twitter",
			"url": "https://twitter.com/danieldogeanu"
		},
		{
			"name": "GitHub",
			"url": "https://github.com/danieldogeanu"
		},
		{
			"name": "LinkedIn",
			"url": "https://www.linkedin.com/in/danieldogeanu/"
		},
		{
			"name": "Behance",
			"url": "https://behance.net/danieldogeanu"
		}
	],
	"shortcuts": [
		{
			"key": "s",
			"description": "Selects Search Input"
		},
		{
			"key": "esc",
			"description": "Clears Search Input / Exits Menu"
		},
		{
			"key": "k",
			"description": "Opens Keyboard Shortcuts Screen"
		},
		{
			"key": "t",
			"description": "Scrolls Back Up"
		}
	]
}
