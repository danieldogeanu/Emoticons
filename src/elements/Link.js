import React from 'react';
import Icon from './Icon';

const Link = (props) => (
	<a href={props.data.url}
		target="_blank"
		data-testid="Link"
		rel="noopener noreferrer">
		{props.social ? (
			<span className="mynet" data-testid="container">
				<span className="show-for-screen-reader">
					{props.data.name}
				</span>
				<Icon name={props.data.name.toLowerCase()} />
			</span>
		) : (props.data.name)}
	</a>
);

export default Link;
