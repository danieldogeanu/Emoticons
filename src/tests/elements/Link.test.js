import React from 'react';
import {render} from '@testing-library/react';
import {author, social} from '../../details.json';
import Link from '../../elements/Link';

describe('Link Element', () => {

	it('renders author link element properly', () => {
		const linkId = 'Link';
		const {getByTestId} = render(<Link data={author} />);
		const renderedLink = getByTestId(linkId);

		expect(renderedLink).toBeInTheDocument();
		expect(renderedLink).toHaveTextContent(author.name);
		expect(renderedLink).toHaveAttribute('href', author.url);
		expect(renderedLink).toHaveAttribute('rel', 'noopener noreferrer');
		expect(renderedLink).toHaveAttribute('target', '_blank');
		expect(renderedLink).toMatchSnapshot();
	});

	it('renders social link element properly', () => {
		const linkId = 'Link';
		const {name, url} = social[0];
		const {getByTestId, getByText, container} = render(<Link data={social[0]} social="true" />);
		const renderedLink = getByTestId(linkId);
		const renderedWrapper = getByTestId('Wrapper');
		const renderedSRText = getByText(name);
		const renderedIcon = container.querySelector('svg');

		expect(renderedLink).toBeInTheDocument();
		expect(renderedLink).toHaveAttribute('href', url);
		expect(renderedLink).toHaveAttribute('rel', 'noopener noreferrer');
		expect(renderedLink).toHaveAttribute('target', '_blank');
		expect(renderedLink).toContainElement(renderedWrapper);
		expect(renderedLink).toMatchSnapshot();

		expect(renderedWrapper).toHaveClass('mynet');
		expect(renderedWrapper).toContainElement(renderedSRText);
		expect(renderedWrapper).toContainElement(renderedIcon);

		expect(renderedSRText).toHaveTextContent(name);
		expect(renderedSRText).toHaveClass('show-for-screen-reader');

		expect(renderedIcon).toHaveClass('icon', name.toLowerCase());
	});

});
