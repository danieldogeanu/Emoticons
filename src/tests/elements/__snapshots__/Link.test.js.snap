// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Element renders author link element properly 1`] = `
<a
  data-testid="Link"
  href="https://danieldogeanu.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Daniel Dogeanu
</a>
`;

exports[`Link Element renders social link element properly 1`] = `
<a
  data-testid="Link"
  href="https://twitter.com/danieldogeanu"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    class="mynet"
    data-testid="container"
  >
    <span
      class="show-for-screen-reader"
    >
      Twitter
    </span>
    <svg
      class="icon twitter"
    >
      <use
        xlink:href="#twitter"
      />
    </svg>
  </span>
</a>
`;
